### PR TITLE
Add readthedocs for hipBLAS

### PR DESCRIPTION
**Pull request by rkamd**
resolves # [LWPMLSE-157](https://ontrack-internal.amd.com/browse/LWPMLSE-157)

Summary of proposed changes:
- Created directory struture for docs similar to rocBLAS
- Added doxygen configuration + script files 
- Added scripts and configuration files for sphinx to generate html and pdf files

### CODE REVIEW

File: docs/source/intro.rst
Comment (amcamd) [inline docs/source/intro.rst:20]: typo,  teh -> the

File: docs/source/intro.rst
Comment (amcamd) [inline docs/source/intro.rst:17]: I would remove this "Currently implemented functionality" section. It is duplicate information. The reader can get this information from the doxygen documentation of the functions.

File: docs/source/install.rst
Comment (amcamd) [inline docs/source/install.rst:24]: AFAIK the script install.sh works for Ubuntu, Centos, Fedora, SLES, and Windows. The Ubuntu Comment is old. I think it would be better to replace "build and install hipBLAS on Ubuntu with a single command." with something like "build and install hipBLAS with a single command." It has been tested on the Linux distributions listed in install.sh."  . I am not sure if we are officially supporting Windows. We can add Windows when we officially support it.

File: docs/source/install.rst
Comment (amcamd) [inline docs/source/install.rst:140]: Change "The -d flag to install.sh installs dependencies" to "Use the install.sh flag -d to install dependencies"

File: docs/source/install.rst
Comment (amcamd) [inline docs/source/install.rst:148]: remove "use a distro other than Ubuntu, or"

File: docs/source/install.rst
Comment (amcamd) [inline docs/source/install.rst:163]: remove this section "Build Library Using Script (Ubuntu only)"

File: docs/source/install.rst
Comment (amcamd) [inline docs/source/install.rst:221]: remove this section "Common Build Problems"

File: docs/source/clients.rst
Comment (amcamd) [inline docs/source/clients.rst:12]: The link does not work. replace with 

These two clients can be built by following the instructions in the "Building and Installing" section.

File: docs/source/clients.rst
Comment (amcamd) [inline docs/source/clients.rst:102]: Can Daine confirm it this section is still valid.

Comment (daineAMD) [reply docs/source/clients.rst:102]: This shouldn't be the case, everything should be passing on CUDA now I believe.

File: docs/source/clients.rst
Comment (daineAMD) [inline docs/source/clients.rst:72]: This isn't the case in hipBLAS, we don't have categories.

File: docs/source/clients.rst
Comment (daineAMD) [inline docs/source/clients.rst:92]: We also don't have the same scheme for datatypes. It would be more like `./hipblas-test --gtest_filter=*axpy*float/*` for this

**Comment by rkamd** [comment]
> I would remove the (url) as extraneous. 

I will remove the link

**Comment by cgmb** [comment]
You might find some of these changes interesting: https://github.com/ROCmSoftwarePlatform/rocSOLVER/pull/318

File: docs/source/clients.rst
Comment (daineAMD) [reply docs/source/clients.rst:12]: Looks like this link is still here and still doesn't link correctly.

File: library/include/hipblas.h
Comment (daineAMD) [inline library/include/hipblas.h:2068]: rocBLAS supports crot and zrot as well (batched and strided_batched as well).

**Review by daineAMD — APPROVED** [review]
Looks good, maybe wait for some other approvals as well as there is a lot here.

**Comment by rkamd** [comment]
> This is not inside a doxygen section correct, I would think it should be ?   Maybe you should set up the build of docs like rocSOLVER so we can verify the results?       E.g. how do your bullets of supported precisions show up? 

Functions.rst file has the information regarding the notations.

File: docs/source/clients.rst
Comment (rkamd) [reply docs/source/clients.rst:12]: The link will start working once this PR is merged to develop branch.

File: docs/source/intro.rst
Comment (amcamd) [inline docs/source/intro.rst:8]: I think the following is better "AMD CPU library that is a thin interface to either AMD **roc** or Nvidia **cu** libraries."

File: docs/source/install.rst
Comment (amcamd) [inline docs/source/install.rst:152]: remove this CMake 3.5 section. PR#427 has code to check that the user has CMake version >= 3.16.8. This CMake 3.5 section is out of date.

**Review by amcamd — APPROVED** [review]
LGTM, thanks for addressing comments

**Pull request merged**
